javaslang/src/main/java/javaslang/control/Try.java

```diff
@@ -55,11 +56,7 @@
      * exception occurs calling {@code supplier.get()}.
      */
     static <T> Try<T> ofSupplier(Supplier<? extends T> supplier) {
-        try {
-            return new Success<>(supplier.get());
-        } catch (Throwable t) {
-            return new Failure<>(t);
-        }
+        return of(supplier::get);
```

Comment:
We need the

```java
Objects.requireNonNull(supplier, "supplier is null");
```

everywhere because of the `::get` call. Otherwise it will throw a NPE before we enter `of(CheckedSupplier)`...